pandas/io/pytables.py

```diff
@@ -3078,83 +3069,6 @@ def write(self, obj, **kwargs):
         self.attrs.name = obj.name
 
 
-class SparseFixed(GenericFixed):
-    def validate_read(self, kwargs):
-        """
-        we don't support start, stop kwds in Sparse
-        """
-        kwargs = super().validate_read(kwargs)
-        if "start" in kwargs or "stop" in kwargs:
-            raise NotImplementedError(
-                "start and/or stop are not supported in fixed Sparse reading"
-            )
-        return kwargs
-
-
-class SparseSeriesFixed(SparseFixed):
-    pandas_kind = "sparse_series"
-    attributes = ["name", "fill_value", "kind"]
-
-    def read(self, **kwargs):
-        kwargs = self.validate_read(kwargs)
-        index = self.read_index("index")
-        sp_values = self.read_array("sp_values")
-        sp_index = self.read_index("sp_index")
-        return SparseSeries(
-            sp_values,
-            index=index,
-            sparse_index=sp_index,
-            kind=self.kind or "block",
-            fill_value=self.fill_value,
-            name=self.name,
-        )
```

Comment:
I don't know exactly how pytables IO works, but in principle we could adjust this to read existing data as a Series[sparse]

Reply:
Yeah I was wondering about that. We do have an issue about supporting EAs in IO (HDF5 in particular). How should we handle this?

I suppose the best thing to do is to change those to return Series[sparse] before this PR?

Reply:
That's also an option. But I don't think we need to try to solve the full "EA in HDF5" problem before this PR, since sparse is actually already supported.

Reply:
Can take a look at that

Reply:
> since sparse is actually already supported.

You mean `Series[sparse]` works already? I didn't think it did.

Reply:
No, I meant that SparseSeries already did, but since Series[sparse] is almost an identical replacement (regarding what data is stored in the Series), it should be rather easy to get Series[sparse] working (while getting EAs in general working might be more effort)

Reply:
To close this subitem, reading SparseSeries / SparseDataFrame files is broken on master. So HDF5 support isn't a blocker.